pkg/resourceinterpreter/subresourceinterpreter/subresource.go

```diff
@@ -0,0 +1,78 @@
+package subresourceinterpreter
+
+import (
+	"context"
+
+	autoscalingapi "k8s.io/api/autoscaling/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/scale"
+	"k8s.io/klog/v2"
+
+	configv1alpha1 "github.com/karmada-io/karmada/pkg/apis/config/v1alpha1"
+	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
+)
+
+// SubResourceInterpreter interpret resource with the related subResource.
+type SubResourceInterpreter struct {
+	restMapper  meta.RESTMapper
+	scaleClient scale.ScalesGetter
+}
+
+// NewSubResourceInterpreter return a new SubResourceInterpreter.
+func NewSubResourceInterpreter(restMapper meta.RESTMapper, scaleClient scale.ScalesGetter) *SubResourceInterpreter {
+	return &SubResourceInterpreter{
+		restMapper:  restMapper,
+		scaleClient: scaleClient,
+	}
+}
+
+// HookEnabled tells if any hook exist for specific resource type and operation type.
+func (i *SubResourceInterpreter) HookEnabled(object *unstructured.Unstructured, operation configv1alpha1.InterpreterOperation) bool {
+	switch operation {
+	case configv1alpha1.InterpreterOperationInterpretReplica:
+		return i.enabledInterpretReplica(object)
+	}
+
+	// other operations are not supported currently.
+	return false
+}
+
+// GetReplicas returns the desired replicas of the object as well as the requirements of each replica.
+func (i *SubResourceInterpreter) GetReplicas(object *unstructured.Unstructured) (int32, *workv1alpha2.ReplicaRequirements, bool, error) {
+	klog.V(4).Infof("Get replicas with scale subResource for request object: %v %s/%s.",
+		object.GroupVersionKind(), object.GetNamespace(), object.GetName())
+
+	scaleObj, err := i.getScaleSubResource(object)
+	if err != nil {
+		return 0, nil, false, err
+	}
+
+	return scaleObj.Spec.Replicas, nil, true, nil
+}
+
+func (i *SubResourceInterpreter) enabledInterpretReplica(object *unstructured.Unstructured) bool {
+	_, err := i.getScaleSubResource(object)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			klog.Infof("There is no subresource of resource %v.", object.GroupVersionKind())
```

Comment:
For resources that don't have scale subresources, this log might be too verbose.

Reply:
How about changing the log level to 4?